doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,734 @@
+directories...", "Docker...", "Git..."], then there's minimal value to start by
+filtering to "git c".
+
+#### Default actions
+
+I propose adding the following actions to the Terminal by default:
+
+```json
+{ "command": { "action":"suggestions", "source": "commandHistory", "useCommandline": true } },
+{ "command": { "action":"suggestions", "source": "directoryHistory" } },
+{ "command": { "action":"suggestions", "source": ["local", "tasks", "commandHistory"], "useCommandline": true, "nesting": "disabled" } },
+{ "command": { "action":"suggestions", "source": ["all"], "useCommandline": false, "nesting": "source" } },
+```
+
+These actions are colloquially:
+* Give me suggestions from my recent commands, using what I've typed
+* Give me suggestions of directories I've recently been in
+* _(After [Tasks] are implemented)_ Give me suggestions from recent commands,
+  commands I've saved, and commands for this project. Don't nest any, so they're
+  all in the top-level menu. Use what I've typed already to start filtering. 
+* Just open the Suggestions UI with all suggestions sources, and group them by
+  the source of the suggestions. 
+
+This should cover most of the basic use cases for suggestions. 
+
+#### Who owns this menu?
+
+There was some discussion of who should own the suggestions menu. The control
+itself? Or the app hosting the control?
+
+A main argument for hosting this UI in the control itself is that any consumer
+of the `TermControl` should be able to display the [shell-driven autocompletion]
+menu. And they should get the UI from us "for free". Consumers shouldn't need to
+reimplement it themselves. In this scenarioThis probably could be done without
+many changes
+* Instead of operating on `Command`s and actions from the terminal settings,
+  the control could just know that all the entries in the menu are "send
+  input" "actions".
+* The control could offer a method to manually invoke the Suggestions UI for a
+  list of {suggestion, name, description} objects.
+* The app layer could easily translate between sendInput actions and these
+  pseudo-actions.
+
+A big argument in favor of having the app layer host the control: Consider an
+app like Visual Studio. When they embed the control, they'll want to style the
+shell-completions UI in their own way. They already have their own intellisense
+menu, and their own UI paradigm.
+
+For now, we'll leave this as something that's owned by the app layer. When we
+get around to finalizing the [shell-driven autocompletion] design, we can
+iterate on ideas for supporting both consumers that want to use a pre-built
+suggestions control, or consumers who want to bring their own.
+
+## Tenets
+
+<table>
+
+<tr><td><strong>Compatibility</strong></td><td>
+
+This shouldn't break any existing flows. This is a general purpose UI element,
+to be extended in a variety of ways. Those customizations will all be opt-in by
+the user, so I'm not expecting any breaking compatibility changes here.

```

Comment:
Gentle disagree, for shell-driven menus - when shells onboard, we *will* see some issues. It's not as general-purpose as you might think.